src/apps/js_generic/js_generic.cpp

```diff
@@ -436,6 +436,54 @@ namespace ccfapp
     return JS_ThrowTypeError(ctx, "Cannot call set on read-only map");
   }
 
+  static JSValue js_kv_map_foreach(
+    JSContext* ctx, JSValueConst this_val, int argc, JSValueConst* argv)
+  {
+    auto map_view =
+      static_cast<KVMap::TxView*>(JS_GetOpaque(this_val, kv_map_view_class_id));
+
+    if (argc != 1)
+      return JS_ThrowTypeError(
+        ctx, "Passed %d arguments, but expected 1", argc);
+
+    JSValue func = argv[0];
+
+    if (!JS_IsFunction(ctx, func))
+      return JS_ThrowTypeError(ctx, "Argument must be a function");
+
+    bool failed = false;
+    map_view->foreach([ctx, func, &failed](const auto& k, const auto& v) {
+      JSValue args[2];
+      args[0] = JS_NewArrayBufferCopy(ctx, k.data(), k.size());
```

Comment:
Let's keep the order as in JS's Map: value, key.

Reply:
Sure, I've actually made it `(value, key, map)` to match [`Map.prototype.forEach`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach), though I haven't included `thisArg`.